zrml/orderbook/src/tests.rs

```diff
@@ -80,12 +116,410 @@ fn fill_order_fails_with_wrong_scoring_rule(scoring_rule: ScoringRule) {
         }));
 
         assert_noop!(
-            Orderbook::fill_order(RuntimeOrigin::signed(ALICE), order_id, None),
+            Orderbook::fill_order(RuntimeOrigin::signed(BOB), order_id, None),
             Error::<Runtime>::InvalidScoringRule
         );
     });
 }
 
+#[test_case(MarketStatus::Proposed; "proposed")]
+#[test_case(MarketStatus::Suspended; "suspended")]
+#[test_case(MarketStatus::Closed; "closed")]
+#[test_case(MarketStatus::CollectingSubsidy; "collecting subsidy")]
+#[test_case(MarketStatus::InsufficientSubsidy; "insufficient subsidy")]
+#[test_case(MarketStatus::Reported; "reported")]
+#[test_case(MarketStatus::Disputed; "disputed")]
+#[test_case(MarketStatus::Resolved; "resolved")]
+fn fill_order_fails_if_market_status_not_active(status: MarketStatus) {
```

Comment:
I think I put this comment in the wrong place. Sorry. Let me try again.

What I mean is that `remove_order` doesn't require that the market have scoring rule `OrderBook`, while `fill_order` does. In both cases you could argue that it is not necessary to have this requirement because `place_order` won't allow placing orders for markets with the wrong scoring rule to begin with. That's why you have to use `mutate_market` to create a scenario which can't happen in the first place just to test this error.

I think both approaches are valid, but we should stick with one approach for both functions.

Reply:
I still don't get it fully yet. 

So, should we remove the code testing a scenario that wouldn't occur? Or should we add a scoring rule check for remove_order?

Reply:
I would also just remove the check from `fill_order`, because `place_order` prevents this. However I would add a debug_assert in fill order.

Reply:
53ef53e1197067a5642e8dd3b7d550b17289555a

Reply:
011547635e993322cf4be2e5bde34c6a7a7b0790